### ISSUE TITLE
Implement deterministic DynamicSceneState fusion for selected CF-GT observations

### ISSUE THREAD

**Issue by JanDuchscherer104** [JanDuchscherer104/ARIA-NBV#89]
TL;DR: the rich Q_H carrier now exposes only a strictly causal selected CF-GT depth prefix. Implement the next, separate state-builder layer that deterministically fuses that prefix into surface, free, unknown, support, recency, and directional memory without mixing raw observations and derived state.\n\nContext and proof:\n- The current thesis requires a static root context plus causal dynamic memory and says a reader must emit either the typed selected-observation prefix or derived DynamicSceneState, never silently both: https://github.com/JanDuchscherer104/ARIA-NBV/blob/f7ca9c86be57fb904f482cc8b1681dbd7fdff0da/docs/typst/thesis/sections/04-method/04-01-scene-representation-requirements.typ#L41-L66\n- Selected depth is bound to selected successor-state history and must not admit unselected candidate renders: https://github.com/JanDuchscherer104/ARIA-NBV/blob/f7ca9c86be57fb904f482cc8b1681dbd7fdff0da/docs/typst/thesis/sections/04-method/04-03-candidate-and-replay-contract.typ#L37-L69\n\nAcceptance criteria:\n- Define one typed DynamicSceneState with explicit CF-GT source, occupied/free/unknown, support, recency, and directional-memory semantics.\n- Consume only selected-observation prefixes; prove state t cannot read current or future selected depth.\n- Preserve missing-modality masks; do not fabricate RGB, DINO, detector, or EVL evidence at unvisited poses.\n- Add deterministic fusion, source-provenance, no-future-observation, and held-out state-transition regressions.\n- Compare this derived-state reader with the raw prefix as distinct named experiment protocols.

**Comment by JanDuchscherer104** [comment]
## External-substitution boundary for S2 — freeze semantics before selecting a sparse backend

The scorer-stack review supports keeping S2 as a **state-contract problem first** and a storage/kernel problem second. Neither Kaolin nor nerfacc should define the representation semantics.

### Required transparent control

Implement the first `S2-ray` control with ordinary PyTorch tensors and a deterministic sparse key, for example quantized root/world coordinates plus a sorted Morton or lexicographic index. Each cell/ray record must distinguish at least:

```text
observed surface / occupied evidence
observed free evidence
unknown (absence of observation, not a stored zero)
support count / weight
recency or last-update step
source role: CF-GT | sensor-like | actor-observed
directional evidence
```

The selected-observation prefix remains the sole update source; state `t` may consume only selected observations `j < t`. Candidate, target, horizon, rewards, all-candidate renders, and oracle labels stay outside the update.

### Backends worth benchmarking after that control passes

**NVIDIA Kaolin SPC/octree**
- Project/docs: https://kaolin.readthedocs.io/en/latest/modules/kaolin.ops.spc.html
- Source: https://github.com/NVIDIAGameWorks/kaolin
- Mechanism worth testing: compact octree/SPC indexing, point-to-octree conversion, batched ray tracing/query operations.
- Integration boundary: replace only the sparse storage/query backend behind the typed `DynamicSceneState`; do not change update semantics or candidate-query outputs.
- Risks: experimental API surface, Torch/CUDA wheel coupling, conversion overhead at the very small horizons used here, and possible nondeterministic GPU reductions.

**nerfacc occupancy grids**
- Methodology: https://nerfacc.readthedocs.io/en/latest/methodology/sampling.html
- Source: https://github.com/nerfstudio-project/nerfacc
- Current conclusion: do **not** adopt as the S2 state. Its occupancy estimator is designed to skip empty samples in a learned density/radiance field. A binary density cache does not preserve causal observed-free versus unobserved-unknown evidence, source role, or exact selected-observation update semantics.
- It may later serve only as a ray-sampling accelerator after an explicit adapter proves semantic parity.

### Smallest decisive benchmark

On one frozen train/validation fixture and one representative CUDA batch, compare the transparent PyTorch control against each backend using identical selected prefixes and candidate queries:

```text
bitwise/tolerance update parity after every selected observation
no-future-observation and source-dropout tests
occupied/free/unknown confusion by cell
candidate-frustum query parity
point/ray permutation and duplicate behavior
p50/p95 update and query latency
peak VRAM and bytes per state
restart/device reproducibility
```

Do not promote a backend unless it reproduces the transparent state/query contract and improves either p95 latency or peak memory materially (predeclare a threshold such as >=20%) on actual `H_max`, point count, and candidate-table shapes. Kernel speed on a synthetic large octree is not sufficient.
